tests/handlers/test_users.py

```diff
@@ -368,7 +371,7 @@ def test_changing_email_ends_any_other_sessions_of_current_user(self):
 
             # force the old `user_id`, simulating that the user is logged in from another browser
             with c.session_transaction() as sess:
-                sess["user_id"] = previous
+                sess["_user_id"] = previous
             rv = self.get_request("/api/users/{}".format(self.factory.user.id))
 
             self.assertEqual(rv.status_code, 404)
```

Comment:
> We should probably try to fix it, as it's testing something important (session/identity behaviour). How is the unit test failing?

The problem lies here: in the new version of "click," using the old user_id for simulating login will not fetch the user's information. However, it will retrieve the latest user_id from the cookie and consequently enable a successful login.

https://github.com/maxcountryman/flask-login/blob/0.6.2/src/flask_login/login_manager.py#L362-L376

Reply:
In the old version of flask_login, the implementation was such that it would only read from the cookie when the `user_id` was not present in the session.

https://github.com/maxcountryman/flask-login/blob/0.4.1/flask_login/login_manager.py#L346-L353

Reply:
@wlach I attempted to resolve this issue by utilizing a different test_client to avoid reusing cookies.

https://github.com/getredash/redash/pull/6138/commits/ea7f59fdb26fb51803436f5cfc514839f1581f3d

Reply:
This is a big project, please give me some time.